File: frontend/src/api/approval.ts
import { invoke } from "@tauri-apps/api/core";

export type ApprovalContext = {
  pending: boolean;
};

export type CommandError = {
  code: string;
  message: string;
};

export const getApprovalContext = () =>
  invoke<ApprovalContext>("get_approval_context");

export const submitApprovalDecision = () =>
  invoke<void>("submit_approval_decision");

File: frontend/src-tauri/src/commands/approval.rs
use serde::Serialize;

use super::CommandError;

#[derive(Debug, Serialize)]
pub struct ApprovalContext {
    pending: bool,
}

#[tauri::command]
pub fn get_approval_context() -> ApprovalContext {
    ApprovalContext { pending: false }
}

#[tauri::command]
pub fn submit_approval_decision() -> Result<(), CommandError> {
    Err(CommandError::new(
        "NO_PENDING_APPROVAL",
        "No approval request is pending.",
    ))
}
